test/include/dlaf_test/util_tile.h

```diff
@@ -0,0 +1,93 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+#include <sstream>
+#include "gtest/gtest.h"
+#include "dlaf/tile.h"
+
+namespace dlaf_test {
+namespace tile_test {
+using namespace dlaf;
+
+/// @brief Sets the element of the tile.
+/// The (i, j)-element of the tile is set to el({i, j}).
+/// @pre el argument is an index of type const TileElementIndex&.
+/// @pre el return type should be T.
+template <class T, class Func>
+void set(Tile<T, Device::CPU>& tile, Func el) {
+  for (SizeType j = 0; j < tile.size().cols(); ++j) {
+    for (SizeType i = 0; i < tile.size().rows(); ++i) {
+      TileElementIndex index(i, j);
+      tile(index) = el(index);
+    }
+  }
+}
+
+namespace internal {
+/// @brief Checks the elements of the tile.
+/// comp(el({i, j}), (i, j)-element) is used to compare the elements.
+/// err_message(el({i, j}), (i, j)-element) is printed for the first element which does not fulfill the comparison.
+/// @pre el argument is an index of type const TileElementIndex&.
+/// @pre comp should have two arguments and return true if the comparison is fulfilled and false otherwise.
+/// @pre err_message should have two arguments and return a string.
+/// @pre el return type should be the same as the type of the first argument of comp and of err_message.
+/// @pre The second argument of comp should be either T, T& or const T&.
+/// @pre The second argument of err_message should be either T, T& or const T&.
+template <class T, class Func1, class Func2, class Func3>
+void check(Tile<T, Device::CPU>& tile, Func1 el, Func2 comp, Func3 err_message, const char* file,
+           const int line) {
+  for (SizeType j = 0; j < tile.size().cols(); ++j) {
+    for (SizeType i = 0; i < tile.size().rows(); ++i) {
+      TileElementIndex index(i, j);
+      if (!comp(el(index), tile(index))) {
+        ADD_FAILURE_AT(file, line) << "Error at index (" << i << ", " << j
+                                   << "): " << err_message(el({i, j}), tile({i, j})) << std::endl;
+        return;
+      }
+    }
+  }
+}
+}
+
+/// @brief Checks the elements of the tile (exact equality).
+/// The (i, j)-element of the tile is compared to el({i, j}).
+/// @pre el argument is an index of type const TileElementIndex&.
+/// @pre el return type should be T.
+template <class T, class Func>
+void checkEQ(Tile<T, Device::CPU>& tile, Func el, const char* file, const int line) {
+  auto comp = [](T expected, T value) { return expected == value; };
+  auto err_message = [](T expected, T value) {
+    std::stringstream s;
+    s << "expected " << expected << " == " << value;
+    return s.str();
+  };
+  internal::check(tile, el, comp, err_message, file, line);
```

Comment:
FYI, you may use [`std::equal_to<T>{}`](https://en.cppreference.com/w/cpp/utility/functional/equal_to) instead of the custom defined lambda `comp`